### ISSUE TITLE
[modern] how to mutate a paginated connection?

### ISSUE THREAD

**Issue by steobrien** [facebook/relay#2067]
Really enjoying Relay Modern so far, thanks to the contributors for a great framework!

### Example:
- There exists an items connection
- It's a paginated list, ordered by creation date, oldest first
- Currently, the application is showing the first 20 items, or 2 pages, out of 10 pages total
- We want to create a new item, by firing off a mutation
- (This new item should ultimately appear at the very end of the list)

### Question:

What should the content of this mutation's `updater` function look like?

If we retrieve the items connection and add to it with `insertEdgeAfter`, my understanding and experience is that it will insert this new item at the end of the _visible_ items, which is incorrect.

Instead, should we repopulate the connection from scratch somehow? If so, how should this be implemented?

**Comment by yuzhi** [comment]
Thanks for trying out Relay. You should be able to use `insertEdgeAfter` on a connection. You can get a connection with RelayConnectionHandler.getConnection with the @connection(key) that you are using in the component. If you are not using the @connection directive, you should be able to modify the `edges` field directly with by modify `getLinkedRecords`, update the array and then `setLinkedRecords`

**Comment by steobrien** [comment]
Thanks for the response, @yuzhi.

I am indeed using the `@connection` directive.

However, for the example I gave, where we've paginated the oldest 2 pages out of 10 total, and have just added a new item through mutation (which should theoretically be on page 10/11), simply using `insertEdgeAfter` will lead to inaccurate ordering – the new item will wrongly appear at the end of page 2.

I could do something imperative, like go check `hasMore` on the `PaginationContainer`, and only `insertEdgeAfter` if that's false, but that feels against the declarative spirit of Relay.
